### PR TITLE
Throttle ttl

### PR DESCRIPTION
**Pull request by shlomi-noach**
fixes https://github.com/github/freno/issues/25 and https://github.com/github/freno/issues/32

This PR introduces:

- throttle TTL
- throttle ratio

new throttle API is:

```
/throttle-app/:app/:ttlMinutes/:ratio
```

Examples:

- `/throttle-app/myapp/20/1` will completely reject requests (`StatusCode 417`) coming from `myapp` for a duration of `20` minutes
- `/throttle-app/myapp/20/0.9` will _mostly_ reject requests (`StatusCode 417`) coming from `myapp` for a duration of `20` minutes. Roughly `9` out of `10` requests will be denied, and roughly `1` out of `10` will be approved. Approval/rejection is based on random dice roll.

**UPDATE**

API further changed in this PR, the basic `/throttle-app` call looks like:

- `/throttle-app/<app-name>/ttl/<ttlMinutes>/ratio/<ratio>`

variations include:

- `/throttle-app/<app-name>/ttl/<ttlMinutes>`
- `/throttle-app/<app-name>/ratio/<ratio>`
- `/throttle-app/<app-name>`

### CODE REVIEW

**Comment by shlomi-noach** [comment]
TBD:

- [ ] shorter API with default values
- [ ] documentation

**Comment by shlomi-noach** [comment]
This runs well in production. I suggest API documentation to be in a followup PR since it needs to document other changes anyhow.

**Comment by shlomi-noach** [comment]
OK updated `http.md` anyhow. All set.

File: go/http/api.go [hunk 1]
```diff
@@ -147,11 +150,34 @@ func (api *APIImpl) AggregatedMetrics(w http.ResponseWriter, r *http.Request, ps
...
-// ThrottleApp forcibly marks given app as throttled. Future requests by this app will be denied.
+// ThrottleApp forcibly marks given app as throttled. Future requests by this app may be denied.
 func (api *APIImpl) ThrottleApp(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
 	appName := ps.ByName("app")
-	err := api.consensusService.ThrottleApp(appName)
+	var expireAt time.Time // default zero
+	var ttlMinutes int64
+	var ratio float64
+	var err error
+	if ps.ByName("ttlMinutes") == "" {
```
Comment (miguelff) [inline go/http/api.go]: Is this consistant with the API docs? 

```
- `/throttle-app/archive/0/0.3`: if already throttled, maintain same TTL and change ratio to `0.3` (`30%` refused). If not already throttled, TTL is one hour
```

Should we have a more explicit way to keep TTL while changing the ratio? Either different endpoints or a more explicit path argument?

Comment (shlomi-noach) [inline go/http/api.go]: good idea. I'll think of something

**Comment by shlomi-noach** [comment]
@miguelff this is ready to be reviewed again. API has changed to be clearer and unambiguous. See https://github.com/github/freno/pull/33/files#diff-562d20151ae48952e3ba3a966fa36308

This is deployed and runs well.

File: go/http/api.go (hunk 1 above)
Comment (shlomi-noach) [inline go/http/api.go]: new API now available for `/throttle-app`

**Pull request merged**